hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -387,31 +387,37 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 	}
 
 	reportLimitedSuportEnabled(hcluster)
-	reportSilecedAlerts(hcluster)
+	reportSilencedAlerts(hcluster)
 	reportProxyConfig(hcluster)
 
 	// If deleted, clean up and return early.
 	if !hcluster.DeletionTimestamp.IsZero() {
 		reportHostedClusterGuestCloudResourcesDeletionDuration(hcluster)
+		reportHostedClusterDeletionDuration(hcluster, r.Clock)
 
 		// Keep trying to delete until we know it's safe to finalize.
 		completed, err := r.delete(ctx, hcluster)
 		if err != nil {
 			return ctrl.Result{}, fmt.Errorf("failed to delete hostedcluster: %w", err)
 		}
+
 		if !completed {
 			log.Info("hostedcluster is still deleting", "name", req.NamespacedName)
 			return ctrl.Result{RequeueAfter: clusterDeletionRequeueDuration}, nil
 		}
+
 		// Now we can remove the finalizer.
 		if controllerutil.ContainsFinalizer(hcluster, finalizer) {
 			controllerutil.RemoveFinalizer(hcluster, finalizer)
 			if err := r.Update(ctx, hcluster); err != nil {
+				hcmetrics.StopReporting(hcluster)
+
 				return ctrl.Result{}, fmt.Errorf("failed to remove finalizer from hostedcluster: %w", err)
 			}
 		}
 
 		reportHostedClusterDeletionDuration(hcluster, r.Clock)
+		hcmetrics.StopReporting(hcluster)
```

Comment:
I'm not following the need for this function. How is current service dealing with remaining metrics from deleted clusters?